### PR TITLE
Fix for not being able to select water tiles in the editor

### PR DESCRIPTION
**Pull request by Interrupt**
- Main issue here was that collision triangles for the water and waterfall tesselated meshes were not being added to the triangle spatial hash
- This was because when triangles are in the spatial hash, they can get decals projected on them
- Switched our triangle collision system to use a new type called CollisionTriangle that can have set collision types
- Projected decals will filter out CollisionTriangles that have the collisionType of Water

### CODE REVIEW

**Comment by Interrupt** [comment]
select-water

File: Dungeoneer/src/com/interrupt/dungeoneer/collision/CollisionTriangle.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.interrupt.dungeoneer.collision;
+
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.utils.Triangle;
+
+public class CollisionTriangle extends Triangle {
+
+    public enum TriangleCollisionType { WORLD, WATER };
```
Comment (PythooonUser) [inline Dungeoneer/src/com/interrupt/dungeoneer/collision/CollisionTriangle.java:8]: Would `LAVA` also need to be an option here? I would think that decals aren't allowed/make sense here either.

Comment (PythooonUser) [reply Dungeoneer/src/com/interrupt/dungeoneer/collision/CollisionTriangle.java:8]: Tested. Works fine with lava as well!

**Review by joshuaskelly — APPROVED** [review]
Looks good!

**Pull request merged**